docs/spec.rst

```diff
@@ -0,0 +1,501 @@
+effect** to be the one located at the *nearest containing scope*
+surrounding the type usage.
+
+-  Class members are contained by classes, which may be contained by
+   other class members or classes, and top-level classes are contained
+   by packages, which may be contained by modules.
+-  If no such defaulting annotation exists, then no defaulting
+   annotation is in effect.
+
+We call any type usage that itself carries a `structurally
+legal <#structural-legality-type-use>`__ type-use annotation
+**explicitly annotated**.
+
+Parameterized types
+~~~~~~~~~~~~~~~~~~~
+
+This section directly builds on `JLS
+4.5 <https://docs.oracle.com/javase/specs/jls/se14/html/jls-4.html#jls-4.5>`__
+to extend nullability to parameterized types.
+
+Parametric nullability
+^^^^^^^^^^^^^^^^^^^^^^
+
+If a type parameter’s bound is nullable, then unannotated usages of that
+type parameter (inside the scope of ``@DefaultNonNull``) have
+*parametric nullability*.
+
+Type arguments of parameterized types
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Tools may wish to check that type arguments are subtypes of the
+corresponding bounds’ types (in addition to base type well-formedness,
+see `JLS
+4.5 <https://docs.oracle.com/javase/specs/jls/se14/html/jls-4.html#jls-4.5>`__).
+As usual, tools may still wish to allow (with warning if desired)
+unsound type arguments involving unspecified nullness.
+
+Specifically, a tool might reject a parameterized type with an explicit
+annotation, such as ``ImmutableList<@Nullable String>``, if
+``ImmutableList``\ ’s type parameter is bounded to be non-null. The
+toool might report an error when encountering this case in source code
+and otherwise ignore the explicit ``@Nullable`` annotation.
+
+Overriding
+~~~~~~~~~~
+
+If a method overrides other methods according to Java language rules
+(see `JLS
+8.4.8.1 <https://docs.oracle.com/javase/specs/jls/se14/html/jls-8.html#jls-8.4.8.1>`__,
+also cf. \ `JVMS
+5.4.5 <https://docs.oracle.com/javase/specs/jvms/se14/html/jvms-5.html#jvms-5.4.5>`__),
+then tools may wish to check that:
+
+-  The overriding method’s augmented return type is be
+   return-type-substitutable for the
+   `supermethods <https://docs.google.com/document/d/1KQrBxwaVIPIac_6SCf--w-vZBeHkTvtaqPSU_icIccc/edit#bookmark=id.5nvbughni6vx>`__\ ’
+   return types (corresponds to covariant return types for base types,
+   an uncontroversial Java feature).
+
+As usual, tools may wish to allow (with warning if desired) unsound
+declarations involving unspecified nullness. Note that
+`supermethods <https://docs.google.com/document/d/1KQrBxwaVIPIac_6SCf--w-vZBeHkTvtaqPSU_icIccc/edit#bookmark=id.5nvbughni6vx>`__
+and
+`superparameters <https://docs.google.com/document/d/1KQrBxwaVIPIac_6SCf--w-vZBeHkTvtaqPSU_icIccc/edit#bookmark=id.m2gxs1ddzqwp>`__
+may be defined by members of parameterized supertypes.
+
+Intended “meaning”
+~~~~~~~~~~~~~~~~~~
+
+The broad intention of nullability is that type usages specified as
```

Comment:
we should probably also reference the definitions of nullable and unspecified nullness in the glossary from this section, and/or change the heading to more specifically discuss non-nullness and field initialization

Reply:
Oops, I did half of this. I added links to those definitions (though at the top of the doc), but I didn't change the section title. I'll do that now ("Concerns around uninitialized objects") and then push again.

Reply:
Done.